pkg/yurttunnel/agent/agent.go

```diff
@@ -73,11 +79,25 @@ func GetTunnelServerAddr(clientset kubernetes.Interface) (string, error) {
 	return fmt.Sprintf("%s:%d", ip.String(), tcpPort), nil
 }
 
-// RunAgent runs the yurttunnel-agent
+// RunAgent runs the yurttunnel-agent which will try to connect yurttunnel-server
 func RunAgent(
 	tlsCfg *tls.Config,
-	serverAddr,
+	tunnelServerAddr,
 	nodeName string,
-	stopChan <-chan struct{}) error {
-	return errors.New("NOT IMPLEMENT YET")
+	stopChan <-chan struct{}) {
+	dialOption := grpc.WithTransportCredentials(credentials.NewTLS(tlsCfg))
+	cc := &anpagent.ClientSetConfig{
+		Address:                 tunnelServerAddr,
+		AgentID:                 nodeName,
+		SyncInterval:            5 * time.Second,
+		ProbeInterval:           5 * time.Second,
+		ReconnectInterval:       5 * time.Second,
+		DialOption:              dialOption,
+		ServiceAccountTokenPath: "",
+	}
+
+	cs := cc.NewAgentClientSet(stopChan)
+	cs.Serve()
+	klog.Infof("start serving grpc request redirected from yurttunel-server: %s",
```

Comment:
the log should be in front of cs.Serve, otherwise it would not be reached before agent exit

Reply:
The `cs.Server()` would not be blocked. The `Server()` will run a goroutine and return. We reply on the stopCh in `YurttunnelAgentOptions.run()` to keep the yurttunnel agent running.